rest-api-model-ql/src/main/kotlin/org/modelix/sample/restapimodelql/ModelQLAPI.kt

```diff
@@ -0,0 +1,97 @@
+        lightModelClientWrapper.runRead {
+            lectureList = LectureList(lectures = allLectures.map { lectureInstance ->
+                Lecture(name = lectureInstance.name,
+                        description = lectureInstance.description,
+                        lectureRef = RouteHelper.urlEncode((lectureInstance.unwrap().reference as LightModelClient.NodeAdapter).nodeId),
+                        room = RouteHelper.urlEncode((lectureInstance.room.unwrap().reference as LightModelClient.NodeAdapter).nodeId),
+                        maxParticipants = lectureInstance.maxParticipants)
+            })
+        }
+        call.respond(lectureList)
+    }
+
+    get<Paths.getLecturesLectureRef> {
+        try {
+            val resolvedLecture: N_Lecture = lightModelClientWrapper.resolveNodeIdToConcept(call.parameters["lectureRef"]!!.decodeURLPart())!! as N_Lecture
+            lateinit var lecture: Lecture
+            lightModelClientWrapper.runRead {
+                lecture = Lecture(name = resolvedLecture.name,
+                        maxParticipants = resolvedLecture.maxParticipants,
+                        lectureRef = RouteHelper.urlEncode((resolvedLecture.unwrap().reference as LightModelClient.NodeAdapter).nodeId),
+                        room = RouteHelper.urlEncode((resolvedLecture.room.unwrap().reference as LightModelClient.NodeAdapter).nodeId),
+                        description = resolvedLecture.description
+                )
+            }
+            call.respond(lecture)
+        } catch (e: RuntimeException) {
+            call.respond(HttpStatusCode.NotFound, "Can not load Lecture: " + e.message)
+        }
+    }
+
+    get<Paths.getRooms> {
+        val allRooms: List<N_Room> = lightModelClientWrapper.getAllRooms()
+        lateinit var roomList: RoomList
+        lightModelClientWrapper.runRead {
+            roomList = RoomList(rooms = allRooms.map { roomInstance ->
+                Room(name = roomInstance.name,
+                        maxPlaces = roomInstance.maxPlaces,
+                        roomRef = RouteHelper.urlEncode((roomInstance.unwrap().reference as LightModelClient.NodeAdapter).nodeId),
+                        hasRemoteEquipment = roomInstance.hasRemoteEquipment)
+            })
+        }
+        call.respond(roomList)
+    }
+
+    get<Paths.getRoomsRoomID> {
+        try {
+            val resolvedRoom: N_Room = lightModelClientWrapper.resolveNodeIdToConcept(call.parameters["roomRef"]!!.decodeURLPart())!! as N_Room
+
+            lateinit var room: Room
+
+            lightModelClientWrapper.runRead {
+                room = Room(name = resolvedRoom.name,
+                        roomRef = RouteHelper.urlEncode((resolvedRoom.unwrap().reference as LightModelClient.NodeAdapter).nodeId),
+                        maxPlaces = resolvedRoom.maxPlaces,
+                        hasRemoteEquipment = resolvedRoom.hasRemoteEquipment
+                )
+            }
+            call.respond(room)
```

Comment:
`runRead` has a return value. Why are these methods written with a `lateinit var`? Shouldn't this work in all cases:

```suggestion
            val room: Room = lightModelClientWrapper.runRead {
                room = Room(name = resolvedRoom.name,
                        roomRef = RouteHelper.urlEncode((resolvedRoom.unwrap().reference as LightModelClient.NodeAdapter).nodeId),
                        maxPlaces = resolvedRoom.maxPlaces,
                        hasRemoteEquipment = resolvedRoom.hasRemoteEquipment
                )
            }
            call.respond(room)
```

Reply:
Yes without the `room = `